packages/flutter/test/material/tabs_test.dart

```diff
@@ -7822,261 +7817,1048 @@ void main() {
+    indicatorLeft = 517.3851539215062;
+    indicatorRight = 607.497721532442;
+    expect(
+      find.byType(TabBar),
+      paints..rrect(
+        rrect: RRect.fromLTRBAndCorners(
+          indicatorLeft,
+          45.0,
+          indicatorRight,
+          48.0,
+          topLeft: const Radius.circular(3.0),
+          topRight: const Radius.circular(3.0),
+        ),
+      ),
     );
-  }, skip: isBrowser); // https://github.com/flutter/flutter/issues/56001
+    expect(indicatorLeft, greaterThan(labelRect.left));
+
+    // Let the animation complete.
+    await tester.pump(const Duration(milliseconds: 200));
 
-  testWidgets('Elastic Tab animation with various size tabs - RTL', (WidgetTester tester) async {
-    final AnimationSheetBuilder animationSheet = AnimationSheetBuilder(
-      frameSize: const Size(800, 100),
+    expect(
+      find.byType(TabBar),
+      paints..rrect(
+        rrect: RRect.fromLTRBAndCorners(
+          labelRect.left,
+          45.0,
+          labelRect.right,
+          48.0,
+          topLeft: const Radius.circular(3.0),
+          topRight: const Radius.circular(3.0),
+        ),
+      ),
     );
-    addTearDown(animationSheet.dispose);
+  });
 
+  testWidgets('Linear Tab animation does not overflow target tab - LTR', (
+    WidgetTester tester,
+  ) async {
     final List<Widget> tabs = <Widget>[
-      const Tab(text: 'Medium'),
-      const Tab(text: 'Extremely Very Long Label'),
-      const Tab(text: 'C'),
+      const Tab(text: 'Short'),
+      const Tab(text: 'A Bit Longer Text'),
+      const Tab(text: 'An Extremely Long Tab Label That Overflows'),
+      const Tab(text: 'Tiny'),
+      const Tab(text: 'Moderate Length'),
+      const Tab(text: 'Just Right'),
+      const Tab(text: 'Supercalifragilisticexpialidocious'),
+      const Tab(text: 'Longer Than Usual'),
     ];
 
     final TabController controller = createTabController(
       vsync: const TestVSync(),
       length: tabs.length,
     );
 
-    Widget target() {
-      return animationSheet.record(
-        boilerplate(
-          textDirection: TextDirection.rtl,
-          child: Container(
-            alignment: Alignment.topLeft,
-            child: TabBar(
-              indicatorAnimation: TabIndicatorAnimation.elastic,
-              controller: controller,
-              tabs: tabs,
-            ),
+    Widget buildTabBar() {
+      return boilerplate(
+        child: Container(
+          alignment: Alignment.topLeft,
+          child: TabBar(
+            indicatorAnimation: TabIndicatorAnimation.linear,
+            controller: controller,
+            tabs: tabs,
           ),
         ),
       );
     }
 
-    await tester.pumpFrames(target(), const Duration(milliseconds: 50));
+    await tester.pumpWidget(buildTabBar());
+
+    await tester.tap(find.byType(Tab).at(2));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 100));
```

Comment:
Just double checking that this first call to `pump()` is necessary here. Here and elsewhere.

Sorry if I already asked about this in the first PR.

Reply:
Initially pump trigger UI update and second pump advances the animation. If we only pump duration, there is no pixel movement.

Reply:
Another example here

https://github.com/flutter/flutter/blob/5d705328dc5fb3a4a1516b18b44324500b371887/dev/benchmarks/microbenchmarks/lib/stocks/build_bench.dart#L29-L30